test/ccc/core/base/panel/legend/legend-panel-spec.js

```diff
@@ -0,0 +1,111 @@
+define([
+  'ccc/pvc',
+  'ccc/def',
+  'test/utils',
+  'test/data-1'
+], function(pvc, def, utils, datas) {
+
+  describe("legend-panel -", function() {
+
+    describe("legendOverflow -", function() {
+      var size = {width: 50, height: 50};
+
+      describe("- Clip Mode", function() {
+
+        it("'legendItemCountMax' isn't taken into account in 'Clip Mode'", function() {
+          var chartOptions = createOptions({legendOverflow: 'clip', legendItemCountMax: 2});
+          var actual = createChart(chartOptions);
+
+          expect(chartOptions.legendItemCountMax < actual.legendItems).toBe(true);
+          expect(actual.isLegendVisible).toBe(true);
+
+        });
+
+        it("'legendSize' will set the size of the legend panel and hide the items if they don't feet it", function() {
+          var chartOptions = createOptions({legendOverflow: 'clip', legendSize: size});
+          var actual = createChart(chartOptions);
+
+          expect(actual.legendItems).toBe(3);
+          expect(actual.isLegendVisible).toBe(true);
+        });
+
+        it("'legendSizeMax' will set the size of the legend panel and hide the items if they don't feet it", function() {
+          var chartOptions = createOptions({legendOverflow: 'clip', legendSize: size});
+          var actual = createChart(chartOptions);
+
+          expect(actual.legendItems).toBe(3);
+          expect(actual.isLegendVisible).toBe(true);
+        });
+
+      });
+
+      describe("- Collapse Mode", function() {
+        it("Should hide the legend panel when the legend items exceed the 'legendItemCountMax' property", function() {
+          var chartOptions = createOptions({legendOverflow: 'collapse', legendItemCountMax: 2});
+          var actual = createChart(chartOptions);
+
+          expect(chartOptions.legendItemCountMax < actual.legendItems).toBe(true);
+          expect(actual.isLegendVisible).toBe(false);
+        });
+
+        it("'legendSize' should hide the legend panel when 1 or more legend items don't feet the size", function() {
+          var chartOptions = createOptions({legendOverflow: 'collapse', legendSize: size});
+          var actual = createChart(chartOptions);
+
+          expect(actual.legendItems).toBe(3);
+          expect(actual.isLegendVisible).toBe(false);
+        });
+
+        it("'legendSizeMax' should hide the legend panel when 1 or more legend items don't feet the size", function() {
+          var chartOptions = createOptions({legendOverflow: 'collapse', legendSize: size});
+          var actual = createChart(chartOptions);
+
+          expect(actual.legendItems).toBe(3);
+          expect(actual.isLegendVisible).toBe(false);
+        });
+      });
+
+    });
+    
+  });
+
+  function createChart(options) {
+    var chart = new pvc.BarChart(options);
+    var dataSpec = datas['relational, series=city|category=date|value=qty, square form'];
+    dataSpec[1].crosstabMode = false;
```

Comment:
The data spec is shared by several unit tests and so cannot be changed this way.
Please create a new data spec if you need to.